src/tools/debugger/FrontendConnector.java

```diff
@@ -209,36 +211,139 @@ public void sendActorHistory() {
       actorObjsToIds.put(a, e.getValue());
     }
 
-    JSONObjectBuilder msg = JsonSerializer.createMessageHistoryJson(messagesPerThread,
-        actorsToIds, actorObjsToIds);
+    JSONObjectBuilder msg = JsonSerializer.createMessageHistoryJson(
+        messagesPerThread, actorsToIds, actorObjsToIds);
 
     String m = msg.toString();
     log("[ACTORS] Message length: " + m.length());
     sender.send(m);
     log("[ACTORS] Message sent?");
     try {
       Thread.sleep(150000);
-    } catch (InterruptedException e1) { }
+    } catch (InterruptedException e1) {
+
+    }
     log("[ACTORS] Message sent waiting completed");
 
     sender.close();
   }
 
   public void requestBreakpoint(final boolean enabled, final URI sourceUri,
-      final int startLine, final int startColumn, final int charLength) {
+      final int startLine, final int startColumn, final int charLength,
+      final String role) {
```

Comment:
Would be good to replace `role` with some enum or so. Passing strings around should generally avoided, I think. Also, having an enum would document what the proper options are.